client/src/components/StreamField/blocks/StreamBlock.js

```diff
@@ -105,29 +134,11 @@ class StreamBlockMenu extends BaseInsertionControl {
   }
 
   renderMenu() {
-    const items = this.groupedChildBlockDefs.map(([group, blockDefs]) => {
-      const groupItems = blockDefs
-        // Allow adding all blockDefs even when disabled, so validation only impedes when saving.
-        // Keeping the previous filtering here for future reference.
-        // .filter((blockDef) => !this.disabledBlockTypes.has(blockDef.name))
-        .map((blockDef) => ({
-          type: blockDef.name,
-          label: blockDef.meta.label,
-          icon: blockDef.meta.icon,
-        }));
-
-      return {
-        label: group || '',
-        type: group || '',
-        items: groupItems,
-      };
-    });
-
```

Comment:
I think it might be nicer to pull this out to a getter or method on its own, it means we may process this data twice but makes the diffs easier to follow and avoids any unexpected changes to the delayed generation of these items.

Something like this...

```js

  get blockItems() {
    return this.groupedChildBlockDefs.map(([group, blockDefs]) => {
      const groupItems = blockDefs
        // Allow adding all blockDefs even when disabled, so validation only impedes when saving.
        // Keeping the previous filtering here for future reference.
        // .filter((blockDef) => !this.disabledBlockTypes.has(blockDef.name))
        .map((blockDef) => ({
          type: blockDef.name,
          label: blockDef.meta.label,
          icon: blockDef.meta.icon,
        }));

      return {
        label: group || '',
        type: group || '',
        items: groupItems,
      };
    });
  }

  renderMenu() {
    ReactDOM.render(
      <ComboBox
        label={comboBoxLabel}
        placeholder={comboBoxLabel}
        items={this.blockItems}
        getItemLabel={(type, item) => item.label}
        getItemDescription={(item) => item.label}
        getSearchFields={(item) => [item.label, item.type]}
        noResultsText={comboBoxNoResults}
        onSelect={this.onSelectBlock.bind(this)}
      />,
      this.combobox,
    );
  }
```